### ISSUE TITLE
`Safe.get_contract()` is replaced with `Safe.contract` cached_property in safe-eth-py 4.4.0

### ISSUE THREAD

**Issue by penandlim** [banteg/ape-safe#43]
https://github.com/safe-global/safe-eth-py/commit/4f1e009f0e6ad23363b16dbc7a3052b3a73a3c1a

**Comment by poolpitako** [comment]
@penandlim want to do a PR with the fix?

**Comment by penandlim** [comment]
So I went down the rabbit hole of trying to find the dependency versions for ape-safe and it seems like it requires specific versions of `web3` and `eth-brownie` packages to successfully use it, and supporting safe-eth-py 4.4.0 would require us to ditch istanbul hardforks / ganache v6.

Couple reasons that cause this is

1. By default brownie uses istanbul hardfork, which is not supported by web3>5.24.0

This causes such that we cannot use web3>5.24.0 due to the istanbul blocks not returning `baseFeePerGas`

```python
 File "brownie/_cli/run.py", line 51, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 110, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/2022_09_28_66_bridge_1MM_SDL_arbitrum.py", line 38, in main
    multisig.preview(safe_tx)
  File "/Users/limjs/IdeaProjects/ape_safe/ape_safe.py", line 342, in preview
    return self.preview_tx(safe_tx, events=events, call_trace=call_trace)
  File "/Users/limjs/IdeaProjects/ape_safe/ape_safe.py", line 320, in preview_tx
    payload = tx.w3_tx.build_transaction()
  File "web3/contract.py", line 1099, in build_transaction
    return build_transaction_for_function(
  File "web3/contract.py", line 1672, in build_transaction_for_function
    prepared_transaction = fill_transaction_defaults(web3, prepared_transaction)
  File "cytoolz/functoolz.pyx", line 249, in cytoolz.functoolz.curry.__call__
  File "web3/_utils/transactions.py", line 114, in fill_transaction_defaults
    default_val = default_getter(web3, transaction)
  File "web3/_utils/transactions.py", line 64, in <lambda>
    web3.eth.max_priority_fee + (2 * web3.eth.get_block('latest')['baseFeePerGas'])
  File "web3/datastructures.py", line 51, in __getitem__
    return self.__dict__[key]  # type: ignore
KeyError: 'baseFeePerGas'
Terminating local RPC client...
```


2. safe-eth-py 4.4.0 requires web3>5.29.2 

web3 was updated sometime 7 months ago and has changed some of the function names to be more pythonic.
https://github.com/ethereum/web3.py/commit/0896611f140fdd8e2737dcc076012d7908e08e38


So seems like the best way is to stay with istanbul forks and older versions of web3 and safe-eth-py.

**Comment by banteg** [comment]
fixed in #48
